nginx/server.conf

```diff
@@ -1,27 +1,47 @@
 server {
     location /.ghost/activitypub {
+        if ($http_x_forwarded_proto = "https") {
+            set $real_scheme https;
+        }
+        if ($http_x_forwarded_proto = "") {
+            set $real_scheme http;
+        }
```

Comment:
Not a blocker - I think we could use a [map directive](https://nginx.org/en/docs/http/ngx_http_map_module.html#map) and dry this up:

```
map $http_x_forwarded_proto $real_scheme {
    default http;
    https https;
}
 ```
 
 I think it might have to go at the `http` level though 🤔
 
 ```
 http {
    map $http_x_forwarded_proto $real_scheme {
        default http;
        https https;
    }

    server {
        location /.ghost/activitypub {
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $real_scheme;

            proxy_pass http://activitypub:8080;
        }

        
        ...
    }
}
 ```

Reply:
Yeah this is nice - I'll try it and update